frontend/public/components/app.jsx

```diff
@@ -158,64 +139,63 @@ class App extends React.PureComponent {
         <GlobalNotifications />
         <Switch>
           <Route path={['/all-namespaces', '/ns/:ns',]} component={RedirectComponent} />
-          <Route path="/overview/all-namespaces" exact component={ClusterOverviewPage} />
-          <Route path="/overview/ns/:ns" exact component={ClusterOverviewPage} />
+          <LazyRoute path="/overview/all-namespaces" exact loader={() => import('./cluster-overview' /* webpackChunkName: "cluster-overview" */).then(m => m.ClusterOverviewPage)} />
```

Comment:
There isn't any risk because TypeScript will complain that `module does not exist` if you use the wrong path. Module imports are statically typed as well, so it will also complain if you try and access a component that is not defined in a module.

Reply:
Thinking on this a bit, contemplating if it tightens up the `<LazyRoute />`/improves readability or not.

Reply:
This is what we had to do in [`resource-pages.ts`](https://github.com/openshift/console/pull/358/files#diff-a1e4f1ecb408f8040c7a324aca6725aaR51). Keep in mind you can't use variables as the path in `import()` calls because it needs to be statically defined for Webpack to construct the chunk graph.

Reply:
Works for me